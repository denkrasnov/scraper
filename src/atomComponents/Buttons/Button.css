.Button {
  min-width: 100px;
  padding: 12px;
  border-radius: 4px;
  background-color: #5A6169;
  color: #fff;
  font-weight: 700;
}
