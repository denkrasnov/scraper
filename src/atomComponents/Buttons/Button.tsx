import React, { FC } from "react";
import { ButtonProps } from "./types";

const Button: FC<ButtonProps> = props => {
  const { children, onClick } = props;
  return (
    <button type="button" onClick={onClick}>
      {children}
    </button>
  );
};

export default Button;
