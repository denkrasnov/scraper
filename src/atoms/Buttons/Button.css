.Button {
  min-width: 80px;
  height: 48px;
  padding: 12px;
  border: none;
  border-radius: 4px;
  background-color: var(--main-red);
  color: var(--white);
  font-weight: 700;
  font-size: 14px;
  box-shadow: 2px 2px 3px var(--shadow-color);
}

/* Implement on desktop */
/* .Button:hover {
  background-color: var(--button-hover);
} */

.Button:active {
  background-color: var(--main-red);
  box-shadow: none;
}

.Button:focus {
  outline: none;
}
