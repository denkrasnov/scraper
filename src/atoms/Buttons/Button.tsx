import React, { FC } from "react";

import { ButtonProps } from "./types";
import styles from "./Button.css";

const Button: FC<ButtonProps> = props => {
  const { disabled, children, onClick, type = "button" } = props;
  return (
    <button
      className={styles.Button}
      disabled={disabled}
      onClick={onClick}
      type={type}
    >
      {children}
    </button>
  );
};

export default Button;
