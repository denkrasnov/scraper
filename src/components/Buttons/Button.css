.Button {
  min-width: 100px;
  padding: 12px;
  border-radius: 4px;
  background-color: var(--button-secondary);
  color: var(--white);
  font-weight: 700;
}
