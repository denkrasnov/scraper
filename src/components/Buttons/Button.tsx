import React, { FC } from "react";
import { Props } from "./types";

const Button: FC<Props> = props => {
  const { children, onClick } = props;
  return (
    <button type="button" onClick={onClick}>
      {children}
    </button>
  );
};

export default Button;
