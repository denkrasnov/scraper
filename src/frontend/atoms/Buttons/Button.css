.root {
  min-width: 80px;
  height: 40px;
  padding: 12px;
  border: none;
  border-radius: 40px;
  background-color: var(--button);
  color: var(--white);
  font-weight: 700;
  font-size: 14px;
  box-shadow: 2px 2px 3px var(--shadow-color);
}

.root:active {
  background-color: var(--main-red);
  box-shadow: none;
}

.root:focus {
  outline: none;
}

.transparent {
  min-width: 80px;
  height: 48px;
  padding: 12px;
  border: 1px solid var(--main-red);
  border-radius: 4px;
  background-color: var(--button-transparent);
  color: var(--main-red);
  font-weight: 700;
  font-size: 14px;
  box-shadow: 2px 2px 3px var(--shadow-color);
}
