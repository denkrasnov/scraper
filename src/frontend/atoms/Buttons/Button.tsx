import { FC } from "react";

import mapPropsToStyles from "../helpers/mapPropsToStyles";
import { ButtonProps } from "./types";
import styles from "./Button.css";

const Button: FC<ButtonProps> = (props) => {
  const { disabled, children, onClick, type = "button", ...restProps } = props;
  const classes = mapPropsToStyles(restProps, styles);

  return (
    <button
      className={classes}
      disabled={disabled}
      onClick={onClick}
      type={type}
    >
      {children}
    </button>
  );
};

export default Button;
