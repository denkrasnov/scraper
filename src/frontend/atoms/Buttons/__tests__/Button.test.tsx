import React from "react";
import { shallow } from "enzyme";

import Button from "..";

describe("Button", () => {
  const children = "__BUTTON_NAME__";
  it("should render", () => {
    const component = shallow(<Button>{children}</Button>);

    expect(component).toMatchSnapshot();
  });

  it("should render with transparent", () => {
    const component = shallow(<Button transparent>{children}</Button>);

    expect(component).toMatchSnapshot();
  });
});
