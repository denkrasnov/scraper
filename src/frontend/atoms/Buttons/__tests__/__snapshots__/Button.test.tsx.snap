// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render 1`] = `
<button
  className="root"
  type="button"
>
  __BUTTON_NAME__
</button>
`;

exports[`Button should render with transparent 1`] = `
<button
  className="root transparent"
  type="button"
>
  __BUTTON_NAME__
</button>
`;
